ecl/hqlcpp/hqlttcpp.cpp

```diff
@@ -9970,6 +9983,52 @@ Datasets
 
   */
 
+
+//Recursively search for an expression which refers to the first selector that is unresolved.
+static IHqlExpression * findOriginalReference(IHqlExpression * * location, IHqlExpression * expr, IHqlExpression * transformed, IHqlExpression * search)
+{
+    for(;;)
+    {
+        if (transformed == search)
+            return expr;
+
+        //If this is a no_select expression that is not resolving from a row then we have a match
+        node_operator transformOp = transformed->getOperator();
+        if ((transformOp == no_select) && !isNewSelector(transformed))
+            return expr;
+
+        //If tree has been transformed to a different structure then do not continue
+        if (transformOp != expr->getOperator())
+            return nullptr;
+
+        //Keep track of the best location that we have seen so far
+        IHqlExpression * symbol = queryLocation(expr);
+        if (symbol && location && (symbol->getStartLine() != 0))
+            *location = symbol;
+
+        //Find the first child expression that is also dependent on that selector
+        unsigned match = NotFound;
```

Comment:
minor: NotFound is a constant from jarray header file, so should really be used for purposes other than related to jarray. (Also, NotFound is of type aindex_t.  It works but mixing types isn't ideal)

Reply:
picky, but fair.  I'll look at changing.